Src/Web/Web.Net45/AspNetDiagnosticTelemetryModule.cs

```diff
@@ -0,0 +1,211 @@
+﻿namespace Microsoft.ApplicationInsights.Web
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Web;
+    using Microsoft.ApplicationInsights.Common;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+    using Microsoft.ApplicationInsights.Web.Implementation;
+
+    /// <summary>
+    /// Listens to ASP.NET DiagnosticSource and enables instrumentation with Activity: let ASP.NET create root Activity for the request.
+    /// </summary>
+    public class AspNetDiagnosticTelemetryModule : IObserver<DiagnosticListener>, IDisposable, ITelemetryModule
+    {
+        private const string AspNetListenerName = "Microsoft.AspNet.TelemetryCorrelation";
```

Comment:
Yes, the only point of the module to parse headers and report request start/stop.
we had a long discussion with Asp.Net team on naming and RequestTracking was one of the suggestions. There is already something in ASP.NET that called request tracing or tracking and this name must not be used.
TelemetryCorrelation was chosen because it explains high-level goal of the whole feature it serves for.